pkg/adapter/nimbus-kubearmor/manager/manager.go

```diff
@@ -81,6 +80,7 @@ func Run(ctx context.Context) {
 func reconcileKsp(ctx context.Context, kspName, namespace string, deleted bool) {
```

Comment:
In the watcher/kspwatcher.go, please use generation to check if the spec is changed in the UpdateFunc

Reply:
Since the watcher will capture any update event even if it's only metadata or status subresource that has changed and we're only interested in the spec, so here we're not checking generation.

Reply:
Will fix in separate PR

Reply:
Here is the [PR](https://github.com/5GSEC/nimbus/pull/64).